addon/helpers/href-to.js

```diff
@@ -21,11 +22,22 @@ function getParamsForGenerateURL(params) {
 }
 
 export default Helper.extend({
+  router: null,
+
+  init() {
+    this._super(...arguments);
+    this.router = getOwner(this).lookup('router:main');
+  },
+
   compute(params, namedArgs) {
     if (namedArgs.params) {
       return hrefTo(this, namedArgs.params);
     } else {
       return hrefTo(this, params);
     }
-  }
+  },
+
+  routerCurrentStateChange: observer('router.currentState', function() {
```

Comment:
As we already access the `"service:-routing"` in `hrefTo`, perhaps we might be able to use something in it? If so, we could avoid another lookup and use a higher level primitive. Perhaps https://api.emberjs.com/ember/release/classes/RouterService/events/routeDidChange might be useful?

Reply:
I initially tried using the event but the version of ember that the add on is using doesn't have the router service events.

Reply:
oh yeah, we're currently running `"ember-source": "~3.4.8"` which is quite old. We should likely upgrade and support whatever version introduced the routing service

Reply:
You want that as part of this PR? I can add it.

Reply:
Nice one, thanks for offering. Perhaps in another PR which we can land first before this one?

Reply:
Given the [LTS Schedule](https://jmorano-dropshare.s3.amazonaws.com/Screen-Shot-2020-07-30-17-27-16.64.png), what version of Ember should we go with? I was thinking 3.8 but that's cutting it close to losing support.

Reply:
I think we can upgrade to the latest version. For anyone running a version that doesn't support the router service, we could point them to an older version

Reply:
Ok. Just a heads up I'm not going to get to this till probably Monday next week.

Reply:
Also as part of the update, want me to refactor using Octane patterns?

Reply:
here is the update PR https://github.com/intercom/ember-href-to/pull/133

Reply:
rebased with update and added looking at the router `routeDidChange` event. I still needed a separate getter for the `router` service vs the `routing` service. the former has the event handling and the latter has the `generateURL` method.

Reply:
fyi, [`3.6`](https://blog.emberjs.com/2018/12/13/ember-3-6-released.html) seems to have introduced `routeDidChange`